azurerm/internal/services/notificationhub/parse/notification_hub.go

```diff
@@ -0,0 +1,38 @@
+package parse
+
+import (
+	"fmt"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+)
+
+type NotificationHubId struct {
+	ResourceGroup string
+	NamespaceName string
+	Name          string
+}
+
+func NotificationHubID(input string) (*NotificationHubId, error) {
+	id, err := azure.ParseAzureResourceID(input)
+	if err != nil {
+		return nil, fmt.Errorf("parsing Notification Hub ID %q: %+v", input, err)
+	}
+
+	app := NotificationHubId{
+		ResourceGroup: id.ResourceGroup,
+	}
+
+	if app.NamespaceName, err = id.PopSegment("namespaces"); err != nil {
+		return nil, fmt.Errorf("parsing Notification Hub ID %q: %+v", input, err)
+	}
+
+	if app.Name, err = id.PopSegment("notificationHubs"); err != nil {
+		return nil, fmt.Errorf("parsing Notification Hub ID %q: %+v", input, err)
```

Comment:
this should already be wrapped:

```suggestion
		return nil, err
```